### PR TITLE
Feature/backend/notifications api

### PR DESCRIPTION
**Pull request by sapiens-sapide**
* Queues a user notification each time a new email arrives with a TTL of 12 hours.
* Add 2 API endpoints to manage notifications : 
    * GET …/notifications, with 2 optional query params `to` and `from` to filter notifications by date range.
    * DELETE …/notifications, with an optional query param `until` to only delete notifications before the `until` date-time.

2 new tables in cassandra : `notification` to hold users' notifications queues, and `notification_ttl` to store default ttls for notifications. 
A new caliopen cli command (`set_notifs_ttls`) is added to generate a default `notification_ttl` table with 7 default values.

### CODE REVIEW

**Comment by sapiens-sapide** [comment]
[Notification nouveau message](https://trello.com/c/d8IGYj8z/32-notification-nouveau-message)

File: src/backend/brokers/go.emails/inbound.go [hunk 1]
```diff
@@ -151,6 +152,19 @@ func (b *EmailBroker) processInbound(in *SmtpEmail, raw_only bool) {
 				if b.Config.LogReceivedMails {
 					log.Infof("EmailBroker : NATS inbound request successfully handled for user %s : %s", rcptId.String(), (*nats_ack)["message"])
 				}
+
+				notif := Notification{
+					Emitter:   "lmtp",
```
Comment (gdchamal) [inline src/backend/brokers/go.emails/inbound.go]: I'm not agree using lmtp as emitter value. Should be better to use a more generic value instead `message`.

File: src/backend/interfaces/REST/go.server/operations/notifications/notifications.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	"github.com/gin-gonic/gin"
+	swgErr "github.com/go-openapi/errors"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+// GetPendingNotif handles GET /notifications
+// two optional params may be in query : `to` and `from` to narrow the notifications list to a time range.
+func GetPendingNotif(ctx *gin.Context) {
```
Comment (gdchamal) [inline src/backend/interfaces/REST/go.server/operations/notifications/notifications.go]: RetrieveNotification is better for consistency naming

File: src/backend/main/go.main/facilities/Notifications/irc.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
+/*
```
Comment (gdchamal) [inline src/backend/main/go.main/facilities/Notifications/irc.go]: I'm not sure we will use irc as notifier channel.

Then if not implemented then not put it in code, for all protocols

File: src/backend/main/py.main/caliopen_main/notification/core.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+
+class Notification(BaseUserCore):
+    """User Notification core class"""
+
+    _model_class = ModelNotification
+    _pkey_name = "user_id"
+
+
+class Notification_ttl(BaseCore):
```
Comment (gdchamal) [inline src/backend/main/py.main/caliopen_main/notification/core.py]: Camel case for class name

File: src/backend/main/py.main/caliopen_main/notification/store.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    user_id = columns.UUID(primary_key=True)
+    timestamp_ = columns.DateTime(primary_key=True)
+    id = columns.UUID(primary_key=True)
+    emitter = columns.Text()
+    type = columns.Ascii()
+    reference = columns.Text()
+    body = columns.Blob()
+
+
+class Notification_ttl(BaseModel):
```
Comment (gdchamal) [inline src/backend/main/py.main/caliopen_main/notification/store.py]: Camel case for class name

File: src/backend/tools/py.CLI/caliopen_cli/cli.py [hunk 6]
```diff
@@ -58,6 +58,11 @@ def main(args=sys.argv):
     sp_shell = subparsers.add_parser('shell')
     sp_shell.set_defaults(func=shell)
 
+    sp_set_ttls = subparsers.add_parser('set_notifs_ttls',
```
Comment (gdchamal) [inline src/backend/tools/py.CLI/caliopen_cli/cli.py]: need update README.md documentation to not forget to use this command during development setup

File: src/backend/brokers/go.emails/inbound.go (hunk 1 above)
Comment (sapiens-sapide) [reply src/backend/brokers/go.emails/inbound.go]: We discussed this wording with @iamdey and thougth that "message" is too much generic ;-)  Email, DM, SMS whatever else, are "message" but they may be handled differently by the front end depending of from which connectors they are coming from : smtp, xmpp, etc.  
Giving a more specific indication of the emitter is helpful for the frontend to handle the message accordingly.

Comment (gdchamal) [reply src/backend/brokers/go.emails/inbound.go]: OK let's go for lmtp (smtp is maybe better than an internal protocol, no ?)

Comment (sapiens-sapide) [reply src/backend/brokers/go.emails/inbound.go]: ok commit #f21d7f5

File: src/backend/interfaces/REST/go.server/operations/notifications/notifications.go (hunk 2 above)
Comment (sapiens-sapide) [reply src/backend/interfaces/REST/go.server/operations/notifications/notifications.go]: I put "GET" verb within REST interface (GetMessage, GetDevice, GetContact), but "RETRIEVE" within store interface.

File: src/backend/main/go.main/facilities/Notifications/irc.go (hunk 3 above)
Comment (sapiens-sapide) [reply src/backend/main/go.main/facilities/Notifications/irc.go]: ok commit #ad6494e

File: src/backend/tools/py.CLI/caliopen_cli/cli.py (hunk 6 above)
Comment (sapiens-sapide) [reply src/backend/tools/py.CLI/caliopen_cli/cli.py]: commit #911cf89

**Comment by sapiens-sapide** [comment]
At least, a refactoring removed `timestamp` field and make use of UUIDv1 for `notif_id`. It guarantees that notif_id always increase and embed the timestamp within the UUID.

**Pull request merged**